src/mongo/db/pipeline/document_source_skip.cpp

```diff
@@ -84,11 +84,12 @@ Pipeline::SourceContainer::iterator DocumentSourceSkip::doOptimizeAt(
     invariant(*itr == this);
 
     auto nextSkip = dynamic_cast<DocumentSourceSkip*>((*std::next(itr)).get());
-
     if (nextSkip) {
-        _nToSkip += nextSkip->getSkip();
-        container->erase(std::next(itr));
-        return itr;
+        if (std::numeric_limits<long long>::max() - _nToSkip - nextSkip->getSkip() >= 0) {
```

Comment:
Please add a comment here describing that we want to avoid overflowing '_nToSkip'. This only works because we know the skips are non-negative, right? If _nToSkip were negative, it looks like we'd overflow.